examples/tcp-close/client.rs

```diff
@@ -150,4 +151,124 @@ impl TcpClient {
 
         Ok(())
     }
+
+    /// Attempts to close several connections sequentially with the expectation
+    /// that the server will close sockets.
+    pub fn run_sequential_expecting_server_to_close_sockets(&mut self, nclients: usize) -> Result<()> {
+        for i in 0..nclients {
+            // Connect to the server and wait.
+            let sockqd: QDesc = self.libos.socket(AF_INET, SOCK_STREAM, 0)?;
+            let qt: QToken = self.libos.connect(sockqd, self.remote)?;
+            let qr: demi_qresult_t = self.libos.wait(qt, None)?;
+
+            match qr.qr_opcode {
+                demi_opcode_t::DEMI_OPC_CONNECT => {
+                    println!("{} clients connected", i + 1);
+
+                    // pop immediately after connect and wait
+                    let pop_qt: QToken = self.libos.pop(sockqd, None)?;
+                    let pop_qr: demi_qresult_t = self.libos.wait(pop_qt, None)?;
+
+                    match pop_qr.qr_opcode {
+                        demi_opcode_t::DEMI_OPC_POP => {
+                            let sga: demi_sgarray_t = unsafe { pop_qr.qr_value.sga };
+                            let received_len: u32 = sga.sga_segs[0].sgaseg_len;
+                            self.libos.sgafree(sga)?;
+                            // 0 len pop represents socket closed from other side.
+                            assert_eq!(
+                                received_len, 0,
+                                "server should have had closed the connection, but it has not"
+                            );
+                            println!("server disconnected (pop returned 0 len buffer)");
+                        },
+                        demi_opcode_t::DEMI_OPC_FAILED => {
+                            let errno: i32 = qr.qr_ret;
+                            assert_eq!(
+                                errno,
+                                libc::ECONNRESET,
+                                "server should have had closed the connection, but it has not"
+                            );
+                            println!("server disconnected (ECONNRESET)");
+                        },
+                        qr_opcode => panic!("unexpected result (qr_opcode={:?})", qr_opcode),
+                    }
+                },
+                demi_opcode_t::DEMI_OPC_FAILED => panic!("operation failed (qr_ret={:?})", qr.qr_ret),
+                qr_opcode => panic!("unexpected result (qr_opcode={:?})", qr_opcode),
+            }
+
+            // Close TCP socket.
+            self.libos.close(sockqd)?;
+        }
+
+        Ok(())
+    }
+
+    /// Attempts to make several connections concurrently.
+    pub fn run_concurrent_expecting_server_to_close_sockets(&mut self, num_clients: usize) -> Result<()> {
+        let mut q_descs: Vec<QDesc> = Vec::default();
+        let mut q_tokens: Vec<QToken> = Vec::default();
```

Comment:
Our code base uses `qts` for naming this variable. Could you change your code to match the existing pattern?